layer/logged_data/logged_data_destination.py

```diff
@@ -0,0 +1,29 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Optional
+from uuid import UUID
+
+from layer.clients.logged_data_service import LoggedDataClient
+from layer.contracts.logged_data import LoggedData
+
+
+class LoggedDataDestination(ABC):
+    @abstractmethod
+    def get_logged_data(
+        self,
+        tag: str,
+        train_id: Optional[UUID] = None,
+        dataset_build_id: Optional[UUID] = None,
+    ) -> LoggedData:
+        pass
```

Comment:
This seems to have the exact same impl for all 3 subclass. Maybe extract the logic here?